### DISCUSSION TITLE
Support for oauth2

### DISCUSSION THREAD

**Discussion by ruben-herold** [poweradmin/poweradmin#579] [Ideas]
hi,

currently, users are only taken from database or from LDAP. More and more services are using outh2 for authentication.
It would be great to have oauth2 support, or?

**Comment by edmondas** [comment]
This sounds ok, but I'm thinking more about user mapping (from an implementation perspective) to the real user (e.g. by email or special ID). Maybe we don't want to automatically create users without permissions in the system, right?

**Comment by ruben-herold** [comment]
@edmondas I can only speak from entra id. In entra id you can configure which group of users can use which oauth2 endpoint. 

I would prefer to auto create user and put them into a default group with minimum rights after creation. So that an admin can put them into the right group after it.

**Comment by edmondas** [comment]
Related ticket #616 has been added to the next milestone.

**Comment by edmondas** [comment]
@ruben-herold Master branch (v4.1.0) now includes OIDC authentication implementation. Development completed with successful testing against Azure, Google, Okta, and Auth0 providers. Implementation designed for compatibility with all OIDC-compliant providers when properly configured. Current implementation uses permission template mapping, with group-based permissions planned for future release.
